controllers/metal3.io/host_state_machine.go

```diff
@@ -428,7 +428,26 @@ func (hsm *hostStateMachine) handleReady(info *reconcileInfo) actionResult {
 }
 
 func (hsm *hostStateMachine) provisioningCancelled() bool {
-	return hsm.imageProvisioningCancelled() && hsm.customDeployProvisioningCancelled()
+	if hsm.Host.Spec.CustomDeploy != nil {
+		if hsm.Host.Spec.CustomDeploy.Method == "" {
```

Comment:
Not sure why an empty custom deploy struct causes provisioning to be unconditionally cancelled? I think that this should go away and the line above should be changed to:

```
if hsm.Host.Spec.CustomDeploy != nil && hsm.Host.Spec.CustomDeploy.Method != "" {
```

In my view the only relevant thing is the Method, and the checks for nil are only to prevent crashes not something that should provide a subtle user interface to behaviour changes.